hack/update-btf

```diff
@@ -1,7 +1,7 @@
 #!/usr/bin/env bash
 set -euo pipefail
 
-BTFHUB_COMMIT=807c5c41894143513100ac58fa07e72df221baa1
+BTFHUB_COMMIT=9d50bd1fb3cc00fca591711f15702cec35347915
```

Comment:
@saschagrunert upstream `807c5c41894143513100ac58fa07e72df221baa1` no longer exists, so had to update this. was this a commit from the main branch? The repo now has no commits older than 14h, so it is concerning from a security standpoint to depend on a project that "rewrites its history". I will raise an issue upstream.